dangerzone/gui/main_window.py

```diff
@@ -51,9 +50,15 @@ def __init__(self, dangerzone: DangerzoneGui) -> None:
         header_layout.addWidget(header_label)
         header_layout.addStretch()
 
-        # Waiting widget, replaces content widget while container runtime isn't available
-        self.waiting_widget = WaitingWidget(self.dangerzone)
-        self.waiting_widget.finished.connect(self.waiting_finished)
+        if isinstance(self.dangerzone.isolation_provider, isolation_provider.Container):
+            # Waiting widget replaces content widget while container runtime isn't available
+            self.waiting_widget: WaitingWidget = WaitingWidgetContainer(self.dangerzone)
+            self.waiting_widget.finished.connect(self.waiting_finished)
+
+        elif isinstance(self.dangerzone.isolation_provider, isolation_provider.Dummy):
+            # Don't wait with dummy converter
+            self.waiting_widget = WaitingWidget()
+            self.dangerzone.is_waiting_finished = True
```

Comment:
Why is this distinction necessary? It would be nice if we could check the code's behavior if `IsolationProvider.install()` fails.

For instance, I just realized that we have this code:

https://github.com/freedomofpress/dangerzone/blob/84b8212e5d85de7a1daf3d522d4f48402a1cbe00/dangerzone/container.py#L83-L85

but we don't check for `False` either in `cli.py` or `main_window.py`. Having an easy way to trigger this code path in our test via the `Dummy` provider I believe would help.

Reply:
Yes. I'm aware: https://github.com/freedomofpress/dangerzone/issues/193. But I feel it may be out of scope for this PR. What do you think?

Reply:
... oh. I think at some point we should though, since we have this code path fresh. I'll add this issue in 0.4.1.

Reply:
And now I see that you have added it in 0.5.0. Ok, let's leave it there.